_posts/2020-01-24-my-favorite-artists.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: "My Favorite Artists And You Can Too!"
+date: 2020-01-24 21:02:00 -0500
+category: blog
+author: Aaron
+author_twitter: shamshirz
+image: /assets/spotify.png
+---
+
+Sharing my recent Spotify 🎧 favorites with Elm and AWS
+![Spotify Logo]({{ "/assets/spotify/spotify.png" | relative_url }}){:class="excerpt-image"}
+
+
+<!--more-->
+I love Spotify. My music taste can often give a better summary of myself than I can. Conveniently, they also have a nice API that can make sharing that with your friends pretty easy too 😏
+
+This year a friend and I shared past playlists as a way to catch up and realized they were a great way to journal. The best way to share that time and experience with each other was to listen to the playlists on our own and at our own pace. This was a surprising, no-pressure way to share myself and I wanted to explore it even more.
+
+## Live Sharing
+
+How can I share a snapshot of myself, without it becoming dated, and without it being an overwhelming task? So far, I had the core data about me - Spotify - and the idea that I wanted it to stay current. There is a very suitable API for the task that will give a list of your most listened to artists over a time range. I found this to be a great starting point. I broke down my wild ideas into a project scoped for a reasonable amount of time and ended up with a serverless API to fetch my top Spotify artists plus a UI widget for the blog to display it 👇. [See how on our Github](https://github.com/SylverStudios/spotify-highlights)
+
+{% include spotifyArtists.html %}
+
+
+## How it Works
+
+The structure is a Lambda wrapper around the Spotify API. I used an API gateway to expose the lambda, and a DynamoDB table to store the Api token. This way, the only origin that can hit the API is our website, and the lambda knows how to handle authenticating with Spotify! That decoupled the serverless-side from the UI, which is a tiny Elm app that knows how to request and decode from the AWS endpoint.
```

Comment:
You may get some value out of splitting out function vs. form, or perhaps what i'm saying is need vs. implementation. I was thinking you could start with high level "something needs to hit the spotify API, meaning something needs to get a spotify access token and hit the spotify API with it" then thereafter talk about the implementation detail of using Lambda and DynamoDB, and the benefits of those as implementation details

Reply:
also I think the API gateway note here is maybe the answer to my CORS question below? I _personally_ would love expansion on this point, maybe others would too.

Reply:
Very good point. I tried to split it up, but ended up making it a little more verbose. Will probs need another pass, but splitting it into: **What do I want** and then **How can we do that** seems like a great move for clarity.